workshop/exercise-6/README.md

```diff
@@ -1 +1,117 @@
+
+### Virtual Services
+A [VirtualService](https://istio.io/docs/reference/config/istio.networking.v1alpha3.html#VirtualService) rule defines a set of traffic routing rules to apply when a host is addressed. Each routing rule defines matching criteria for traffic of a specific protocol. If the traffic is matched, then it is sent to a named [destination](https://istio.io/docs/reference/config/istio.networking.v1alpha3.html#Destination) service (or [subset](https://istio.io/docs/reference/config/istio.networking.v1alpha3.html#Subset) or version of it) defined in the service registry.
+
+### Destination Rules
+A [DestinationRule](https://istio.io/docs/reference/config/istio.networking.v1alpha3.html#Destination) defines policies that apply to traffic intended for a service after routing has occurred. These rules specify configuration for load balancing, connection pool size from the sidecar, and outlier detection settings to detect and evict unhealthy hosts from the load balancing pool. Any destination `host` and `subset` referenced in a `VirtualService` rule must have a corresponding `DestinationRule` defined.
+
+### External Services
+An [ExternalService](https://istio.io/docs/reference/config/istio.networking.v1alpha3.html#ExternalService) rule enables services within the mesh to access access an external service not necessarily managed by Istio.  The rule describes the endpoints, ports and protocols of a white-listed set of mesh-external domains and IP blocks that services in the mesh are allowed to access.
+
+## The Guestbook Example
+In the Guestbook example, there is one service: guestbook.  The guestbook service has two distinct versions: the base version (version 1) and the modernized version (version 2).  Each version of the service has three instances based on the number of replicas in [guestbook-deployment.yaml](https://github.com/linsun/examples/blob/master/guestbook-go/guestbook-deployment.yaml) and [guestbook-v2-deployment.yaml](https://github.com/linsun/examples/blob/master/guestbook-go/guestbook-v2-deployment.yaml).  By default, prior to creating any rules, Istio will route requests equally across version 1 and version 2 of the guestbook service.  However, new versions of a service can easily introduce bugs to the service mesh, so following Canary Deployments and A/B Testing are recommended practices to follow.
+
+### Canary Deployment
+In `Canary Deployments`, newer versions of services are incrementally rolled out to minizmize the risk and impact of any bugs introduced by the newer version.  To prevent Istio from performing the default routing behavior between the original and modernized guestbook service version, the following rules should be defined:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: virtual-service-guestbook
+spec:
+  hosts:
+  - guestbook
+  http:
+  - route:
+    - destination:
+        host: guestbook
+        subset: v1
+```
+
```

Comment:
would be good to say this is 100% traffic goes to v1. and the yaml below is round robin between v1 and v2 equally (or based on # of replicas).